lib/src/main/kotlin/controller/Controller.kt

```diff
@@ -0,0 +1,129 @@
+
+
+class Controller {
+
+    private val avlManager = AVLTreeManager()
+    private val rbManager = RBTreeManager()
+    private val binManager = BINTreeManager()
+
+
+    var tree: DrawTree? = null
+
+    fun showFiles(): List<List<String>> {
+        val avlTrees = avlManager.getSavedTreesNames()
+        val rbTrees = rbManager.getSavedTreesNames()
+
+        createDummyFiles(avlTrees, rbTrees)
+
+        return listOf(
+            rbTrees.subList(0, min(3, rbTrees.size)),
+            avlTrees.subList(0, min(3, avlTrees.size)),
+            binManager.getSavedTreesNames()
+        )
+    }
+
+    private fun createDummyFiles(avlTrees: List<String>, rbTrees: List<String>) {
+        val avlPath = System.getProperty("user.dir") + "/saved-trees/AVL-trees"
+        val rbPath = System.getProperty("user.dir") + "/saved-trees/RB-trees"
+
+        File(avlPath).mkdirs()
+        File(rbPath).mkdirs()
+
+        if (avlTrees.isNotEmpty()) {
+            for (name in avlTrees) {
+                File(avlPath, name).run {
+                    createNewFile()
+                }
+            }
+        }
+
+        if (rbTrees.isNotEmpty()) {
+            for (name in rbTrees) {
+                File(rbPath, name).run {
+                    createNewFile()
+                }
+            }
+        }
+
+    }
+
+    fun createTree(treeName: String, id: Int): DrawTree {
+        when (id) {
+            0 -> tree = RBDrawableTree(treeName, rbManager)
+            1 -> tree = AVLDrawableTree(treeName, avlManager)
+            2 -> tree = BINDrawableTree(treeName, binManager)
+        }
+
+        tree?.initTree()
+
+        return tree ?: throw NullPointerException()
+    }
+
+    fun insert(value: String): DrawTree {
+        val key = if (isInt(value.trim())) value.toInt() else value.hashCode()
+
+        tree?.insert(Container(Pair(key, value))) ?: throw NullPointerException()
+        tree?.updateTree() ?: throw NullPointerException()
+        tree?.repositisonTree(800f, 10f) ?: throw NullPointerException()
+
+        return tree ?: throw NullPointerException()
+    }
+
+    fun find(value: String): DrawTree {
+        val key = if (isInt(value.trim())) value.toInt() else value.hashCode()
+
+        tree?.find(key) ?: throw NullPointerException()
+
+        return tree ?: throw NullPointerException()
+
+    }
+
+    fun delete(value: String): DrawTree {
+        val key = if (isInt(value.trim())) value.toInt() else value.hashCode()
+
+        tree?.delete(Container(Pair(key, ""))) ?: throw NullPointerException()
+        tree?.updateTree() ?: throw NullPointerException()
+        tree?.repositisonTree(800f, 10f) ?: throw NullPointerException()
+
+        return tree ?: throw NullPointerException()
+    }
+
+
+    private fun isInt(value: String): Boolean {
+        return try {
+            Integer.parseInt(value.trim())
+            true
+        } catch (ex: NumberFormatException) {
+            false
+        }
+    }
```

Comment:
I think the following looks better. Don't forget to fix usages.
```suggestion
    private fun String.presentAsKey(): Int {
        return this.trim().toIntOrNull() ?: this.hashCode()
    }
```